### PR TITLE
Update Linux build and test workflows to use lit

### PR DESCRIPTION
**Pull request by llvm-beanz**
This change updates the build and test workflows in our azure pipelines and the GitHub action for code coverage data to use lit instead of ctest as the test runner. This adds additional test coverage by running some of the LLVM & Clang tests.

### CODE REVIEW

File: .github/workflows/coverage-gh-pages.yml [hunk 1]
```diff
@@ -33,11 +33,11 @@ jobs:
       - name: Install dependencies
         run: sudo apt install -y ninja-build
       - name: Configure
-        run: mkdir build && cd build && cmake -G Ninja -DLLVM_PARALLEL_LINK_JOBS=1 -DLLVM_USE_LINKER=lld -DCMAKE_BUILD_TYPE=Release -DDXC_COVERAGE=On -C ${{github.workspace}}/cmake/caches/PredefinedParams.cmake -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ ${{github.workspace}}
+        run: mkdir build && cd build && cmake -G Ninja -DLLVM_PARALLEL_LINK_JOBS=1 -DLLVM_USE_LINKER=lld -DCMAKE_BUILD_TYPE=Release -DDXC_USE_LIT=On -DDXC_COVERAGE=On -C ${{github.workspace}}/cmake/caches/PredefinedParams.cmake -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ ${{github.workspace}}
```
Comment (Keenuts) [inline .github/workflows/coverage-gh-pages.yml:36]: I see a lot of `cd build && ninja ...`, they should work with `ninja -C build ...` no?

Comment (llvm-beanz) [reply .github/workflows/coverage-gh-pages.yml:36]: Yea, I could also use `cmake -B <>` to remove the `mkdir && cd` there too.

Comment (llvm-beanz) [reply .github/workflows/coverage-gh-pages.yml]: Updated.

**Review by Keenuts — APPROVED** [review]
thanks!

**Review by pow2clk — APPROVED** [review]
I also approve!

**Pull request merged**